walletclient.go

```diff
@@ -1,117 +1,151 @@
-// Package walletclient is a Go client for interacting with Spv Wallet.
 package walletclient
 
 import (
-	"github.com/bitcoin-sv/spv-wallet-go-client/transports"
+	"net/http"
+
+	"github.com/bitcoin-sv/spv-wallet/models"
 	"github.com/bitcoinschema/go-bitcoin/v2"
 	"github.com/libsv/go-bk/bec"
 	"github.com/libsv/go-bk/bip32"
 	"github.com/libsv/go-bk/wif"
 	"github.com/pkg/errors"
 )
 
-// ClientOps are used for client options
-type ClientOps func(c *WalletClient)
-
-// WalletClient is the spv wallet go client representation.
+// WalletClient is the spv wallet Go client representation.
 type WalletClient struct {
-	transports.TransportService
-	accessKey        *bec.PrivateKey
-	accessKeyString  string
-	transport        transports.TransportService
-	transportOptions []transports.ClientOps
-	xPriv            *bip32.ExtendedKey
-	xPrivString      string
-	xPub             *bip32.ExtendedKey
-	xPubString       string
+	signRequest     *bool
+	server          *string
+	accessKeyString *string
+	xPrivString     *string
+	xPubString      *string
+	httpClient      *http.Client
+	accessKey       *bec.PrivateKey
+	adminXPriv      *bip32.ExtendedKey
+	xPriv           *bip32.ExtendedKey
+	xPub            *bip32.ExtendedKey
 }
 
-// New create a new wallet client
-func New(opts ...ClientOps) (*WalletClient, error) {
-	client := &WalletClient{}
+// NewWalletClientWithXPrivate creates a new WalletClient instance using a private key (xPriv).
+// It configures the client with a specific server URL and a flag indicating whether requests should be signed.
+// - `xPriv`: The extended private key used for cryptographic operations.
+// - `serverURL`: The URL of the server the client will interact with.
+// - `sign`: A boolean flag to determine if the outgoing requests should be signed.
+func NewWalletClientWithXPrivate(xPriv, serverURL string, sign bool) (*WalletClient, error) {
+	return newWalletClient(
+		&WithXPriv{XPrivString: &xPriv},
+		&WithHTTP{ServerURL: &serverURL},
+		&WithSignRequest{Sign: &sign},
+	)
+}
 
-	for _, opt := range opts {
-		opt(client)
-	}
+// NewWalletClientWithXPublic creates a new WalletClient instance using a public key (xPub).
+// This client is configured for operations that require a public key, such as verifying signatures or receiving transactions.
+// - `xPub`: The extended public key used for cryptographic verification and other public operations.
+// - `serverURL`: The URL of the server the client will interact with.
+// - `sign`: A boolean flag to determine if the outgoing requests should be signed.
+func NewWalletClientWithXPublic(xPub, serverURL string, sign bool) (*WalletClient, error) {
+	return newWalletClient(
+		&WithXPub{XPubString: &xPub},
+		&WithHTTP{ServerURL: &serverURL},
+		&WithSignRequest{Sign: &sign},
+	)
+}
```

Comment:
Do we need it? There is no place where `wc.xPub` is used.  It seems that only xPriv and AccessKey are used and in tests you check that client with only xpub return errors

Reply:
not sure but i think `js-client` has it `constructor(serverUrl: string, options: XpubWithoutSigning, loggerConfig?: LoggerConfig);`

Reply:
lest see what others would decide